File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    days = list(math.ceil((100 - i) / j) for i, j in zip(progresses, speeds))
    count = 0
    header = days[0]
    for i in days:
        if i <= header:
            count += 1
        else:
            header = i
            answer.append(count)
            count = 1
    answer.append(count)
    return answer

File: Java/프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> list = new ArrayList<>();
        
        // 초기화
        int dep = (int)Math.ceil((100 - progresses[0])/(float)speeds[0]);;
        int count = 1;
        
        for(int i = 1; i < progresses.length; i++){
            int d = (int)Math.ceil((100 - progresses[i])/(float)speeds[i]);
            
            if(d > dep){
                dep = d;
                list.add(count);
                count = 0;
            }
            count ++;
        }
        
        list.add(count);
        
        // 리스트 > 배열
        int[] answer = new int[list.size()];
        
        if (progresses.length == 0)
            answer[0] = 1;
        
        for(int i = 0; i < list.size(); i++){
            answer[i] = list.get(i);
        }
        
        return answer;
    }
}